### PR TITLE
env_prefix and env_postfix

### PR DESCRIPTION
**Pull request by scopatz**
This came up in https://github.com/pypa/virtualenv/issues/1299 and is probably the right way to implement the env name printing.

@gforsyth - will this mess you up for xonda?

I have based this on #2993, mostly so I don't have redo all of the CI work.  Also I ran black on the prompt subpackage

### CODE REVIEW

File: requirements-tests.txt [hunk 1]
```diff
@@ -1,12 +1,12 @@
 ply
 py
 pytest
+flake8
+pyflakes<2.1.0,>=2.0.0
```
Comment (asottile) [inline requirements-tests.txt]: (sorry for the driveby) -- if you remove `pyflakes` from this file, `flake8` will automatically resolve the correct version (it has ~an equivalent requirements in its `setup.py`)

Comment (scopatz) [reply requirements-tests.txt]: Ahh OK thanks!

Comment (asottile) [reply requirements-tests.txt]: without this, you're hitting a bug in `pip` (which doesn't actually implement dependency resolution)

basically first-pass `pip` greedily sees `pyflakes` and says "latest version of pyflakes please!", then later it sees in the second pass that flake8 wants pyflakes>=2.0,<2.1 and says "I've already seen pyflakes, satisfied!" even when it's a conflict :sob:

Comment (scopatz) [reply requirements-tests.txt]: Yeah.... I figured that was happening. :frowning_face:

**Comment by bskinn** [comment]
Hrm -- I was picturing implementing it such that the prefix and postfix wouldn't need to be explicitly included in the `$PROMPT` markup... that `{env_name}` would pull in `{env_prefix}` and `{env_postfix}` by default.

That would make `$PROMPT` more concise; and I can't picture any scenarios offhand where someone would want to reuse, e.g., `{env_prefix}` in a prompt.

**Comment by scopatz** [comment]
@bskinn - that is what is on the current master now, but I guess you are saying that the `env_name()` function would actually look up `env_prefix` and `env_postfix` in the `$PROMPT_FIELDS` and then evaluate them? 

That is a nice idea... and a little cleaner.

**Comment by bskinn** [comment]
> I guess you are saying that the `env_name()` function would actually look up `env_prefix` and `env_postfix` in the `$PROMPT_FIELDS` and then evaluate them?

Yep, exactly -- the same behavior as now, except that the values are configurable, rather than hard-coded.

**Comment by scopatz** [comment]
OK, got it.  Just pushed up that change.

**Comment by scopatz** [comment]
>  The latest release of conda claims to have xonsh support. Now, that support is broken and doesn't work, but still, we're almost there.

haha yeah, it is probably a good idea to just move xonda into conda :)

**Comment by gforsyth** [comment]
looks good. thanks @scopatz !
